File: nekbone/mxm.f
      subroutine mxm(a,n1,b,n2,c,n3)
      implicit none

      integer i,j,k,n1,n2,n3
      double precision a(n1,n2),b(n2,n3),c(n1,n3)

      do j=1,n3
        do i=1,n1
          c(i,j) = 0.0
          do k=1,n2
            c(i,j) = c(i,j) + a(i,k)*b(k,j)
          enddo
        enddo
      enddo

      return
      end

File: nekbone/mxm.c
void mxm(double a[][], int n1, double b[][], int n2, double c[][], int n3){
    int i, j, k;
    for(j = 0; j < n3; j++){
        for(i = 0; i < n1; i++){
            c[j][i] = 0.0;
            for(k = 0; k < n2; k++){
                c[j][i] = c[j][i] + a[k][i] * b[j][k];
            }
        }
    }
    
    return;
}
